src/dmd/clone.d

```diff
@@ -1081,6 +1083,7 @@ DtorDeclaration buildExternDDtor(AggregateDeclaration ad, Scope* sc)
     ad.members.push(func);
     func.addMember(sc2, ad);
     func.dsymbolSemantic(sc2);
+    func.functionSemantic(); // to infer attributes
```

Comment:
While this inference change makes sense, I'm not sure whether it's a good idea to combine it with a PR that "just adds a test".

Reply:
The commit message already aid otherwise, I've updated the PR description now, too.